evaluator/src/test/kotlin/com/xebia/functional/xef/evaluator/ItemSpecBuilderSpec.kt

```diff
@@ -0,0 +1,165 @@
+        outputResponse { firstOutputResponse }
+        outputResponse { secondOutputResponse }
+      }
+
+    invalidItemSpec shouldBeLeft listOf<ValidationError>(EmptyItemSpecInput)
+  }
+
+  @Test
+  fun shouldInvalidWhenContextDescriptionIsEmpty() = runTest {
+    val inputItem = "Please provide a movie title, genre and director"
+    val myContextDescription = "      "
+    val firstOutputResponse =
+      """
+        | Movie Title: The Pursuit of Dreams
+        | Genre: Drama
+        | Director: Christopher Nolan
+        """
+        .trimIndent()
+    val secondOutputResponse = "I don't know"
+
+    val invalidItemSpec =
+      ItemSpec(inputItem) {
+        contextDescription { myContextDescription }
+        outputResponse { firstOutputResponse }
+        outputResponse { secondOutputResponse }
+      }
+
+    invalidItemSpec shouldBeLeft listOf<ValidationError>(EmptyContextDescription)
+  }
+
+  @Test
+  fun shouldInvalidWhenFirstOutputResponseIsEmpty() = runTest {
+    val inputItem = "Please provide a movie title, genre and director"
+    val myContextDescription = "Contains information about a movie"
+    val firstOutputResponse = "   "
+    val secondOutputResponse = "I don't know"
+
+    val invalidItemSpec =
+      ItemSpec(inputItem) {
+        contextDescription { myContextDescription }
+        outputResponse { firstOutputResponse }
+        outputResponse { secondOutputResponse }
+      }
+
+    invalidItemSpec shouldBeLeft listOf<ValidationError>(EmptyOutputResponse)
+  }
+
+  @Test
+  fun shouldInvalidWhenSecondOutputResponseIsEmpty() = runTest {
+    val inputItem = "Please provide a movie title, genre and director"
+    val myContextDescription = "Contains information about a movie"
+    val firstOutputResponse =
+      """
+        | Movie Title: The Pursuit of Dreams
+        | Genre: Drama
+        | Director: Christopher Nolan
+        """
+        .trimIndent()
+    val secondOutputResponse = " "
+
+    val invalidItemSpec =
+      ItemSpec(inputItem) {
+        contextDescription { myContextDescription }
+        outputResponse { firstOutputResponse }
+        outputResponse { secondOutputResponse }
+      }
+
+    invalidItemSpec shouldBeLeft listOf<ValidationError>(EmptyOutputResponse)
+  }
+
+  @Test
+  fun shouldInvalidWhenAllOutputsResponseAreEmpty() = runTest {
+    val inputItem = "Please provide a movie title, genre and director"
+    val myContextDescription = "Contains information about a movie"
+    val firstOutputResponse = " "
+    val secondOutputResponse = " "
+
+    val invalidItemSpec =
+      ItemSpec(inputItem) {
+        contextDescription { myContextDescription }
+        outputResponse { firstOutputResponse }
+        outputResponse { secondOutputResponse }
+      }
+
+    invalidItemSpec shouldBeLeft listOf<ValidationError>(EmptyOutputResponse, EmptyOutputResponse)
+  }
+
+  @Test
+  fun shouldInvalidWhenAnyCombinationOfInputAreEmpty() = runTest {
```

Comment:
All the invalid tests could be replaced by a property test that generates a tuple of the invalid `ItemSpec` + the error list.